universe/ValueRefs.h

```diff
@@ -664,6 +685,82 @@ template <>
 FO_COMMON_API std::vector<std::string> Variable<std::vector<std::string>>::Eval(const ScriptingContext& context) const;
 
 
+///////////////////////////////////////////////////////////
+// NamedRef                                              //
+///////////////////////////////////////////////////////////
+template <typename T>
+NamedRef<T>::NamedRef(const std::string& value_ref_name) :
+    m_value_ref_name(value_ref_name)
+{
+    DebugLogger() << "ctor(NamedRef<T>): " << typeid(*this).name() << " value_ref_name: " << m_value_ref_name;
+    // FIXME need to check the invariances
+    if (auto ref = GetValueRef()) {
```

Comment:
probably a parser race to init this here...

Reply:
what does "parser race" mean?

Reply:
do the named refs get parsed by the time they are looked up? does anything  in them depend on other stuff being already parsed?

Reply:
the value refs get parsed when parsing happens. 
evaluation happens when the valueref which contains a reference (e.g. via `Lookup`) to the named valueref  is being evaluated.
the registration happens directly after the contained valueref parsed valuer is parsed

for the named-in-the-middle named valuerefs construction etc is basically the same as if you would not name it. 
the top-level named valuerefs are independent of their referencing.
`Lookup` is independent from valueref registration. You might reference a non-existing or not-yet-existing name. 

Everything is completely parsed before any valuerefs are evaluated.